internal/resource/config/incomingproxysettings/incoming_proxy_settings_resource.go

```diff
@@ -0,0 +1,285 @@
+				},
+			},
+			"client_cert_ssl_header_name": schema.StringAttribute{
+				Description: "While the proxy server is configured to pass client certificates as HTTP request headers, specify the header name here.",
+				Computed:    false,
+				Optional:    true,
+			},
+			"client_cert_chain_ssl_header_name": schema.StringAttribute{
+				Description: "While the proxy server is configured to pass client certificates as HTTP request headers, specify the chain header name here.",
+				Computed:    false,
+				Optional:    true,
+			},
+			"proxy_terminates_https_conns": schema.BoolAttribute{
+				Description: "Allows you to globally specify that connections to the reverse proxy are made over HTTPS even when HTTP is used between the reverse proxy and PingFederate.",
+				Computed:    true,
+				Optional:    true,
+			},
+		},
+	}
+
+	id.ToSchema(&schema)
+	resp.Schema = schema
+}
+
+func addOptionalIncomingProxySettingsFields(ctx context.Context, addRequest *client.IncomingProxySettings, plan incomingProxySettingsResourceModel) error {
+
+	if internaltypes.IsDefined(plan.ForwardedIpAddressHeaderName) {
+		addRequest.ForwardedIpAddressHeaderName = plan.ForwardedIpAddressHeaderName.ValueStringPointer()
+	}
+
+	if internaltypes.IsDefined(plan.ForwardedIpAddressHeaderIndex) {
+		addRequest.ForwardedIpAddressHeaderIndex = plan.ForwardedIpAddressHeaderIndex.ValueStringPointer()
+	}
+
+	if internaltypes.IsDefined(plan.ForwardedHostHeaderName) {
+		addRequest.ForwardedHostHeaderName = plan.ForwardedHostHeaderName.ValueStringPointer()
+	}
+
+	if internaltypes.IsDefined(plan.ForwardedHostHeaderIndex) {
+		addRequest.ForwardedHostHeaderIndex = plan.ForwardedHostHeaderIndex.ValueStringPointer()
+	}
+
+	if internaltypes.IsDefined(plan.ClientCertSSLHeaderName) {
+		addRequest.ClientCertSSLHeaderName = plan.ClientCertSSLHeaderName.ValueStringPointer()
+	}
+
+	if internaltypes.IsDefined(plan.ClientCertChainSSLHeaderName) {
+		addRequest.ClientCertChainSSLHeaderName = plan.ClientCertChainSSLHeaderName.ValueStringPointer()
+	}
+
+	if internaltypes.IsDefined(plan.ProxyTerminatesHttpsConns) {
+		addRequest.ProxyTerminatesHttpsConns = plan.ProxyTerminatesHttpsConns.ValueBoolPointer()
+	}
+
+	return nil
+
+}
+
+// Metadata returns the resource type name.
+func (r *incomingProxySettingsResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_incoming_proxy_settings"
+}
+
+func (r *incomingProxySettingsResource) Configure(_ context.Context, req resource.ConfigureRequest, _ *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	providerCfg := req.ProviderData.(internaltypes.ResourceConfiguration)
+	r.providerConfig = providerCfg.ProviderConfig
+	r.apiClient = providerCfg.ApiClient
+
+}
+
+func readIncomingProxySettingsResponse(ctx context.Context, r *client.IncomingProxySettings, state *incomingProxySettingsResourceModel, existingId *string) diag.Diagnostics {
```

Comment:
This method doesn't need to return diagnostics since it never adds any diagnostic values